stl/src/print.cpp

```diff
@@ -59,25 +59,53 @@ namespace {
     public:
         _Allocated_string() = default;
 
-        explicit _Allocated_string(__crt_unique_heap_ptr<wchar_t>&& _Other_str, const size_t _Other_capacity) noexcept
-            : _Str(_STD move(_Other_str)), _Str_capacity(_Other_capacity) {}
+        ~_Allocated_string() {
+            if (_Using_heap()) {
+                _Str.~__crt_unique_heap_ptr<wchar_t>();
+            }
+        }
 
-        [[nodiscard]] wchar_t* _Data() const noexcept {
-            return _Str.get();
+        [[nodiscard]] wchar_t* _Data() noexcept {
+            return _Using_heap() ? _Str.get() : _Buffer;
         }
 
         [[nodiscard]] size_t _Capacity() const noexcept {
             return _Str_capacity;
         }
 
-        void _Reset() noexcept {
+        bool _Grow(const size_t _Capacity) {
+            if (_Capacity <= _Str_capacity) {
+                return true;
+            }
+
             _Str.release();

```

Comment:
(no change requested)

This is not release-the-ownership, this is deallocating-release!

Reply:
Yeah I'll leave a comment on it in source too, cause it is a confusing function name